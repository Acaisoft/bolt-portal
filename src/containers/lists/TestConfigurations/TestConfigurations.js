import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { TestConfigurationsTable } from '~components'
import { List } from '~containers'
import { GET_CONFIGS_QUERY } from '~services/GraphQL/Queries'

export class TestConfigurations extends Component {
  static propTypes = {
    projectId: PropTypes.string,
    onDelete: PropTypes.func,
  }

  render() {
    const { projectId, onDelete, ...listProps } = this.props

    const query = GET_CONFIGS_QUERY

    const variables = {
      projectId,
      order_by: [{ id: 'asc' }],
    }

    return (
      <List
        paginationDataKey="configuration_aggregate"
        query={query}
        variables={variables}
        {...listProps}
      >
        {({ data, loading }) => (
          <TestConfigurationsTable
            configurations={data && data.configuration}
            getDetailsUrl={configuration =>
              `/test-configurations/${configuration.id}`
            }
            loading={loading}
            onDelete={onDelete}
            projectId={projectId}
          />
        )}
      </List>
    )
  }
}

export default TestConfigurations
