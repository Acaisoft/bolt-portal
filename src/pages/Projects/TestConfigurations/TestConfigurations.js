import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Mutation } from 'react-apollo'

import { Typography, withStyles } from '@material-ui/core'

import { DeleteModal } from '~components'
import { TestConfigurationsList } from '~containers/lists'
import styles from './TestConfigurations.styles'

import { DELETE_CONFIG_MUTATION } from '~services/GraphQL/Mutations'
import { GET_CONFIGS_QUERY } from '~services/GraphQL/Queries'

export class TestConfigurations extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    projectId: PropTypes.string,
  }

  state = {
    openDeleteModal: false,
    configName: '',
    deleteConfigId: null,
  }

  handleSubmit = async (id, { delMutation }) => {
    await delMutation({ variables: { id } })
    this.setState({ openDeleteModal: false })
  }

  handleModalOpen = ({ id, name }) => {
    this.setState({ openDeleteModal: true, configName: name, deleteConfigId: id })
  }

  handleModalClose = () => {
    this.setState({ openDeleteModal: false })
  }

  render() {
    const { classes, projectId } = this.props
    const { openDeleteModal, configName, deleteConfigId } = this.state

    return (
      <div className={classes.root}>
        <Typography variant="body2">
          Here you can manage all Test Run Configurations
        </Typography>
        <Mutation
          mutation={DELETE_CONFIG_MUTATION}
          refetchQueries={[{ query: GET_CONFIGS_QUERY, variables: { projectId } }]}
        >
          {(delMutation, { data }) => (
            <DeleteModal
              open={openDeleteModal}
              handleClose={this.handleModalClose}
              handleSubmit={() => this.handleSubmit(deleteConfigId, { delMutation })}
              type="repository"
              name={configName}
            />
          )}
        </Mutation>
        <TestConfigurationsList
          projectId={projectId}
          showPagination
          onDelete={this.handleModalOpen}
        />
      </div>
    )
  }
}

export default withStyles(styles)(TestConfigurations)
