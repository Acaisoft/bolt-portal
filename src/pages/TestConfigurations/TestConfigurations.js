import React from 'react'
import PropTypes from 'prop-types'

import { Redirect, Route, Switch } from 'react-router-dom'

import TestExecutionsPage from '~pages/TestExecutions'
import ListPage from './List'
import DetailsPage from './Details'
import CreateOrEditPage from './CreateOrEdit'

export function TestConfigurations({ match }) {
  return (
    <Switch>
      <Route path={`${match.path}`} exact component={ListPage} />
      <Route path={`${match.path}/create`} exact component={CreateOrEditPage} />
      <Route
        path={`${match.path}/:configurationId`}
        component={ConfigurationSubpages}
      />
      <Redirect from="*" to={match.url} />
    </Switch>
  )
}
TestConfigurations.propTypes = {
  match: PropTypes.shape({
    path: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
  }).isRequired,
}

function ConfigurationSubpages({ match }) {
  return (
    <Switch>
      <Route path={`${match.path}`} exact component={DetailsPage} />
      <Route path={`${match.path}/test-runs`} component={TestExecutionsPage} />
    </Switch>
  )
}

export default TestConfigurations
