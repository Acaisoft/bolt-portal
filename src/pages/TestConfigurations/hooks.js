import { gql } from '@apollo/client'

import { useMutationWithState } from 'hooks'

const RUN_TEST_CONFIGURATION = gql`
  mutation runTestConfiguration($configurationId: UUID!, $coldStart: Boolean) {
    testrun_start(conf_id: $configurationId, no_cache: $coldStart) {
      execution_id
    }
  }
`
export function useConfigurationRun(configurationId) {
  const { loading, mutation, error } = useMutationWithState(RUN_TEST_CONFIGURATION, {
    variables: { configurationId },
  })

  return { loading, mutation, error }
}

const DELETE_TEST_CONFIGURATION = gql`
  mutation deleteTestConfiguration($configurationId: UUID) {
    testrun_configuration_delete(pk: $configurationId) {
      affected_rows
    }
  }
`
export function useConfigurationDelete(configurationId) {
  const { loading, mutation, error } = useMutationWithState(
    DELETE_TEST_CONFIGURATION,
    { variables: { configurationId } }
  )

  return { loading, mutation, error }
}

const CLONE_SCENARIO = gql`
  mutation clone_scenario($configurationId: UUID!) {
    testrun_configuration_clone(configuration_id: $configurationId) {
      returning {
        new_configuration_id
      }
      affected_rows
    }
  }
`

export function useConfigurationClone(configurationId) {
  const { loading, mutation, error } = useMutationWithState(CLONE_SCENARIO, {
    variables: { configurationId },
  })

  return { loading, mutation, error }
}
