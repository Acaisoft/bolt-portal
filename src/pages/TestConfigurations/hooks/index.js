export { default as useConfigurationDelete } from './useConfigurationDelete'
export { default as useConfigurationRun } from './useConfigurationRun'
