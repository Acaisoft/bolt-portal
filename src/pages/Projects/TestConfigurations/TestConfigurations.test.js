import React from 'react'
import { shallow } from 'enzyme'

import { TestConfigurations } from './TestConfigurations'
import { mocks } from '~utils/tests'

jest.mock('~components', () => ({
  DataTable: 'DataTableMock',
  DeleteModal: 'DeleteModalMock',
}))

const initComponent = overrides => {
  const mockProps = {
    classes: {},
  }
  const wrapper = shallow(
    <TestConfigurations classes={mocks.ClassesProxy} {...mockProps} {...overrides} />
  )
  const instance = wrapper.instance()
  return { instance, wrapper }
}

describe('page: TestConfigurations', () => {
  describe('rendering', () => {
    it('should render without crashing', () => {
      const { wrapper } = initComponent()
      expect(wrapper).toBeTruthy()
    })
    it('should render as expected', () => {
      const { wrapper } = initComponent()
      expect(wrapper).toMatchSnapshot()
    })
  })
  describe('events', () => {
    describe('handleSubmit', () => {
      it('should close delete config and close modal', () => {
        const { instance } = initComponent()
        const fakeId = 'fakeID'
        const deleteConfigMock = jest.fn()
        instance.handleSubmit(fakeId, { deleteConfigMock })
        expect(instance.state.openDeleteModal).toBe(false)
      })
    })
    describe('handleModalOpen', () => {
      it('should open modal', () => {
        const { instance } = initComponent()
        const fakeID = 'faleId'
        const fakeName = 'fakeName'
        instance.handleModalOpen(fakeID, fakeName)
        expect(instance.state.openDeleteModal).toBe(true)
      })
      it('should close modal', () => {
        const { instance } = initComponent()
        instance.handleModalClose()
        expect(instance.state.openDeleteModal).toBe(false)
      })
    })
  })
})
