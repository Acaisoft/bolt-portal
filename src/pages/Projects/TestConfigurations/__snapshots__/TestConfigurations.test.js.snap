// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page: TestConfigurations rendering should render as expected 1`] = `
<div>
  <TypographyMock
    variant="body2"
  >
    Here you can manage all Test Run Configurations
  </TypographyMock>
  <Mutation
    mutation={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "deleteConfig",
            },
            "operation": "mutation",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": Object {
                    "kind": "Name",
                    "value": "config",
                  },
                  "arguments": Array [
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "where",
                      },
                      "value": Object {
                        "fields": Array [
                          Object {
                            "kind": "ObjectField",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                            "value": Object {
                              "fields": Array [
                                Object {
                                  "kind": "ObjectField",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "_eq",
                                  },
                                  "value": Object {
                                    "kind": "Variable",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "id",
                                    },
                                  },
                                },
                              ],
                              "kind": "ObjectValue",
                            },
                          },
                        ],
                        "kind": "ObjectValue",
                      },
                    },
                  ],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "delete_configuration",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "returning",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "name",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [
              Object {
                "defaultValue": undefined,
                "directives": Array [],
                "kind": "VariableDefinition",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "uuid",
                  },
                },
                "variable": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "id",
                  },
                },
              },
            ],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 162,
          "start": 0,
        },
      }
    }
    refetchQueries={
      Array [
        Object {
          "query": Object {
            "definitions": Array [
              Object {
                "directives": Array [],
                "kind": "OperationDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "getTestConfigurations",
                },
                "operation": "query",
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [
                        Object {
                          "kind": "Argument",
                          "name": Object {
                            "kind": "Name",
                            "value": "where",
                          },
                          "value": Object {
                            "fields": Array [
                              Object {
                                "kind": "ObjectField",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "project_id",
                                },
                                "value": Object {
                                  "fields": Array [
                                    Object {
                                      "kind": "ObjectField",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "_eq",
                                      },
                                      "value": Object {
                                        "kind": "Variable",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "projectId",
                                        },
                                      },
                                    },
                                  ],
                                  "kind": "ObjectValue",
                                },
                              },
                            ],
                            "kind": "ObjectValue",
                          },
                        },
                        Object {
                          "kind": "Argument",
                          "name": Object {
                            "kind": "Name",
                            "value": "limit",
                          },
                          "value": Object {
                            "kind": "Variable",
                            "name": Object {
                              "kind": "Name",
                              "value": "limit",
                            },
                          },
                        },
                        Object {
                          "kind": "Argument",
                          "name": Object {
                            "kind": "Name",
                            "value": "offset",
                          },
                          "value": Object {
                            "kind": "Variable",
                            "name": Object {
                              "kind": "Name",
                              "value": "offset",
                            },
                          },
                        },
                        Object {
                          "kind": "Argument",
                          "name": Object {
                            "kind": "Name",
                            "value": "order_by",
                          },
                          "value": Object {
                            "kind": "Variable",
                            "name": Object {
                              "kind": "Name",
                              "value": "order_by",
                            },
                          },
                        },
                      ],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "configuration",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "name",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "executions",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "id",
                                  },
                                  "selectionSet": undefined,
                                },
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "start",
                                  },
                                  "selectionSet": undefined,
                                },
                              ],
                            },
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "repository",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "id",
                                  },
                                  "selectionSet": undefined,
                                },
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "url",
                                  },
                                  "selectionSet": undefined,
                                },
                              ],
                            },
                          },
                        ],
                      },
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [
                        Object {
                          "kind": "Argument",
                          "name": Object {
                            "kind": "Name",
                            "value": "where",
                          },
                          "value": Object {
                            "fields": Array [
                              Object {
                                "kind": "ObjectField",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "project_id",
                                },
                                "value": Object {
                                  "fields": Array [
                                    Object {
                                      "kind": "ObjectField",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "_eq",
                                      },
                                      "value": Object {
                                        "kind": "Variable",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "projectId",
                                        },
                                      },
                                    },
                                  ],
                                  "kind": "ObjectValue",
                                },
                              },
                            ],
                            "kind": "ObjectValue",
                          },
                        },
                      ],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "configuration_aggregate",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "aggregate",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "count",
                                  },
                                  "selectionSet": undefined,
                                },
                              ],
                            },
                          },
                        ],
                      },
                    },
                  ],
                },
                "variableDefinitions": Array [
                  Object {
                    "defaultValue": undefined,
                    "directives": Array [],
                    "kind": "VariableDefinition",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "uuid",
                      },
                    },
                    "variable": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "projectId",
                      },
                    },
                  },
                  Object {
                    "defaultValue": undefined,
                    "directives": Array [],
                    "kind": "VariableDefinition",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "Int",
                      },
                    },
                    "variable": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "limit",
                      },
                    },
                  },
                  Object {
                    "defaultValue": undefined,
                    "directives": Array [],
                    "kind": "VariableDefinition",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "Int",
                      },
                    },
                    "variable": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "offset",
                      },
                    },
                  },
                  Object {
                    "defaultValue": undefined,
                    "directives": Array [],
                    "kind": "VariableDefinition",
                    "type": Object {
                      "kind": "ListType",
                      "type": Object {
                        "kind": "NonNullType",
                        "type": Object {
                          "kind": "NamedType",
                          "name": Object {
                            "kind": "Name",
                            "value": "execution_order_by",
                          },
                        },
                      },
                    },
                    "variable": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "order_by",
                      },
                    },
                  },
                ],
              },
            ],
            "kind": "Document",
            "loc": Object {
              "end": 526,
              "start": 0,
            },
          },
          "variables": Object {
            "projectId": undefined,
          },
        },
      ]
    }
  >
    <Component />
  </Mutation>
  <TestConfigurations
    onDelete={[Function]}
    showPagination={true}
  />
</div>
`;
